### PR TITLE
DEP: removed old-style Composable app classes

### PR DESCRIPTION
**Pull request by GavinHuttley**

### CODE REVIEW

File: changelog.d/20230324_162931_Gavin.Huttley.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+### DOC
+
+-   A bullet item for the DOC category.
+
+-->
+<!--
+### Deprecations
+
+-   Removed the original Composable app classes and related decorators for
+    user based apps. `user_function` and `appify` are replaced by the
```
Comment (Nick-Foto) [inline changelog.d/20230324_162931_Gavin.Huttley.md:35]: We can add here:
Removed the original Composable app classes and related decorators for user base apps and replaced with `define_app` decorator.

File: src/cogent3/app/composable.py [hunk 2]
```diff
@@ -129,651 +129,6 @@ def to_json(self):
...
-        super(ComposableModel, self).__init__(**kwargs)
-        discontinued(
-            "class",
-            "ComposableModel",
-            "2023.1",
-            "see developer docs for new class hierarchy",
-        )
-
-
-class ComposableHypothesis(Composable):  # pragma: no cover
```
Comment (Nick-Foto) [inline src/cogent3/app/composable.py:564]: Did we replace all these classes in new decorator model? ComposableHypothesis, ComposableDistance and others. I am not sure about this.

File: changelog.d/20230324_162931_Gavin.Huttley.md (hunk 1 above)
Comment (Nick-Foto) [reply changelog.d/20230324_162931_Gavin.Huttley.md:35]: And what about _checkpointable class?

File: src/cogent3/app/composable.py [hunk 3]
```diff
@@ -129,651 +129,6 @@ def to_json(self):
...
-        super(ComposableDistance, self).__init__(**kwargs)
-        discontinued(
-            "class",
-            "ComposableDistance",
-            "2023.1",
-            "see developer docs for new class hierarchy",
-        )
-
-
-class _checkpointable:
```
Comment (Nick-Foto) [inline src/cogent3/app/composable.py:586]: I am not sure about how we replaced this class in new decorator.

File: tests/test_app/test_composable.py [hunk 4]
```diff
@@ -615,26 +606,6 @@ def test_user_function_multiple():
     assert got_2 == {("s1", "s2"): 2.0, ("s2", "s1"): 2.0}
 
 
-def test_appify():
```
Comment (Nick-Foto) [inline tests/test_app/test_composable.py:618]: I am not sure that we have this test in new decorator.

File: changelog.d/20230324_162931_Gavin.Huttley.md (hunk 1 above)
Comment (GavinHuttley) [reply changelog.d/20230324_162931_Gavin.Huttley.md:35]: Your suggested additional text exists in the file, not sure why github not showing it...

`_checkpointable` is a private class, no-one should have been using it

File: src/cogent3/app/composable.py (hunk 2 above)
Comment (GavinHuttley) [reply src/cogent3/app/composable.py:564]: Absolutely! The `define_app` decorator completely removed the need for subclassing.

File: src/cogent3/app/composable.py (hunk 3 above)
Comment (GavinHuttley) [reply src/cogent3/app/composable.py:586]: All support for checkpointing is now in `app_type=WRITER`.

File: tests/test_app/test_composable.py (hunk 4 above)
Comment (GavinHuttley) [reply tests/test_app/test_composable.py:618]: No longer a valid test for the new decorator.

**Pull request merged**